components/director/pkg/resource/resource.go

```diff
@@ -1,77 +1,179 @@
 	Tombstone Type = "tombstone"
-	// IntegrationSystem missing godoc
+	// IntegrationSystem type represents integration system resource.
 	IntegrationSystem Type = "integrationSystem"
-	// SystemAuth missing godoc
+	// SystemAuth type represents system auth resource.
 	SystemAuth Type = "systemAuth"
-	// FetchRequest missing godoc
+	// FetchRequest type represents generic fetch request resource. This resource does not assume the referenced resource type of the FR.
 	FetchRequest Type = "fetchRequest"
-	// Specification missing godoc
+	// DocFetchRequest type represents document fetch request resource.
+	DocFetchRequest Type = "docFetchRequest"
+	// APISpecFetchRequest type represents API specification fetch request resource.
+	APISpecFetchRequest Type = "apiSpecFetchRequest"
+	// EventSpecFetchRequest type represents Event specification fetch request resource.
+	EventSpecFetchRequest Type = "eventSpecFetchRequest"
+	// Specification type represents generic specification resource. This resource does not assume the referenced resource type of the Spec.
 	Specification Type = "specification"
-	// Document missing godoc
+	// APISpecification type represents API specification resource.
+	APISpecification Type = "apiSpecification"
+	// EventSpecification type represents Event specification resource.
+	EventSpecification Type = "eventSpecification"
+	// Document type represents document resource.
 	Document Type = "document"
-	// BundleInstanceAuth missing godoc
+	// BundleInstanceAuth type represents bundle instance auth resource.
 	BundleInstanceAuth Type = "bundleInstanceAuth"
-	// API missing godoc
+	// API type represents api resource.
 	API Type = "api"
-	// EventDefinition missing godoc
+	// EventDefinition type represents event resource.
 	EventDefinition Type = "eventDefinition"
-	// AutomaticScenarioAssigment missing godoc
+	// AutomaticScenarioAssigment type represents ASA resource.
 	AutomaticScenarioAssigment Type = "automaticScenarioAssigment"
-	// Webhook missing godoc
+	// Webhook type represents generic webhook resource. This resource does not assume the referenced resource type of the Webhook.
 	Webhook Type = "webhook"
-	// Tenant missing godoc
+	// AppWebhook type represents application webhook resource.
+	AppWebhook Type = "appWebhook"
+	// RuntimeWebhook type represents runtime webhook resource.
+	RuntimeWebhook Type = "runtimeWebhook"
+	// Tenant type represents tenant resource.
 	Tenant Type = "tenant"
-	// TenantIndex missing godoc
-	TenantIndex Type = "tenantIndex"
-	// Schema missing godoc
+	// TenantAccess type represents tenant access resource.
+	TenantAccess Type = "tenantAccess"
+	// Schema type represents schema resource.
 	Schema Type = "schemaMigration"
 )
 
-// SQLOperation missing godoc
+var tenantAccessTable = map[Type]string{
+	// Tables
+
+	Application: "tenant_applications",
+	Runtime:     "tenant_runtimes",
+
+	// Views
+
+	RuntimeContext:        "runtime_contexts_tenants",
```

Comment:
have you thought about making this a method on the storage entity type ? sql_types.go/entity

Reply:
Yes initially we were thinking where to put it and the reasoning was that this is a static metadata for the type and not for a specific instance of the entity. It is more of a metadata of the table than of a row of the table.

Reply:
One more thing is that only the create and update operations has an actual entity, all other operations has just a condition. If we add it to the entity we need to provide an empty entity in Get/Select/Delete instead of resourceType right now.

Reply:
ok, that makes sense. Another idea - we can reuse the idea for the top level mapping - we could put the m2m ref table/view name in the repo constuctor (example check func NewRepository(conv APIDefinitionConverter) *pgRepository) - there we already pass some table names to the unified repositories below, we could pass this one also. This way you distribute the cenralized (static) mapping to the locations that care about it and they dont have to look up the static mapping.
In the end you would end up with a repo instantiation that carries all the details (what is the table, what is the tenant isolation/m2m table, is it top level, etc)